drivers/SmartThings/zigbee-switch/src/init.lua

```diff
@@ -85,6 +100,133 @@ local device_init = function(self, device)
+    zcl_body = generic_body.GenericBody(discover_command_payload)
+  })
+  local send_message = messages.ZigbeeMessageTx({
+      address_header = addrh,
+      body = message_body
+  })
+  device:send(send_message)
+end
+
+local function read_attribute_function(device, cluster_id, attr_id)
+  local read_body = read_attribute.ReadAttribute( attr_id )
+  local zclh = zcl_messages.ZclHeader({
+    cmd = data_types.ZCLCommandId(read_attribute.ReadAttribute.ID)
+  })
+
+  local addrh = messages.AddressHeader(
+      zb_const.HUB.ADDR,
+      zb_const.HUB.ENDPOINT,
+      device:get_short_address(),
+      device:get_endpoint(cluster_id.value),
+      zb_const.HA_PROFILE_ID,
+      cluster_id.value
+  )
+  local message_body = zcl_messages.ZclMessageBody({
+    zcl_header = zclh,
+    zcl_body = read_body
+  })
+  return messages.ZigbeeMessageTx({
+    address_header = addrh,
+    body = message_body
+})
+end
+
+local function device_added(driver, device, event)
+  device:set_find_child(find_child)
+  
+  local num_switch_server_eps = 0
+  local main_endpoint = device:get_endpoint(clusters.OnOff.ID)
+  local updated_flag = false
+  
+  send_discover_command(device)
+  for _, ep in ipairs(device.zigbee_endpoints) do
+   
+    num_switch_server_eps = num_switch_server_eps + 1
+    if ep.id ~= main_endpoint then 
+      if updated_flag == false then
+        device:try_update_metadata({profile="basic-switch"})
+        updated_flag = true
+      end
+      if device:supports_server_cluster(clusters.OnOff.ID, ep.id) then
+        if find_child(device, num_switch_server_eps) == nil then
+          local name = string.format("%s %d", device.label, num_switch_server_eps)
+          local child_profile = "basic-switch"
+          driver:try_create_device(
+            {
+              type = "EDGE_CHILD",
+              label = name,
+              profile = child_profile,
+              parent_device_id = device.id,
+              parent_assigned_child_key = string.format("%02X", num_switch_server_eps),
+              vendor_provided_label = name
+            }
+          )
+        end
+      end
+    end
+  end
+  
+  device:refresh()
+end
+
+local function do_payload(rx)  -- use discover attribute response payload to do read attribute payload
+  local out_str = " "
+  for i = 1, #rx.body_bytes do
+    out_str = out_str .. string.format(" %02X", string.byte(rx.body_bytes:sub(i,i)))
+  end
+  local bytes = {}
+  for b in out_str:gmatch("%S+") do
+      table.insert(bytes, b)
+  end
+  table.remove(bytes, 1) -- remove first type which means discover attribute command is completed
+  local result = {}
+  for i = 1, #bytes, 3 do   -- remove data type information
+      local low = bytes[i]
+      local high = bytes[i+1]
+      if high then
+          local value = tonumber(high, 16) * 0x100 + tonumber(low, 16)
+          table.insert(result, value)
+      end
+  end
+  return result
+end
+
+local function all_zigbee_message_handler(self, message_channel)
```

Comment:
This is going to change how EVERY message processed by ANY Zigbee Switch on the platform is handled.  I do not think we should make this change.